### ISSUE TITLE
lineTo issues when lines overlap

### ISSUE THREAD

**Issue by ps-lvuksan** [pixijs/pixi.js#7254]
I've had this weird issue for over a year now and I am trying to fix it now but I am out of ideas. I'm not sure if it's related to Pixi, WebGL or something else. We have an application where we render maps from a xml file using Pixi and we noticed these strange lines extending from what should actually be drawn. The xml file is autogenerated from an image and often has a lot of unnecessary and overlapping lines and polylines in it. Except from this issue, Pixi has handled it extraordinarily well, so thanks!

### Expected Behavior
`lineTo` should not render any additional lines outside of the points I gave it.
### Current Behavior
When I render a polygon (no difference if closed or not), in special circumstances some lines are being extended beyond what should be drawn. The length of the extended line increases with thicker lineStyles.

From what I've observed, it happens when you draw a line in a specific angle `a`, then draw a longer line with angle `-a`, going back and overlapping the previous line and then draw another line in angle `b`. 
### Possible Solution
Unfortunately I have no idea.
### Steps to Reproduce
```
// 2ND POINT GOES BACK AND CAUSES LINE TO EXTEND WEIRDLY
const g = new PIXI.Graphics();
g.lineStyle(2, 0xff0000);

g.moveTo(42, 99);
g.lineTo(35, 92);
g.lineTo(61, 117);
g.lineTo(85, 118);

g.position.set(100, 0);
```
### Environment
We saw it happen on many different devices with different OS and browsers.
Tested on:
- **`pixi.js` version**: 5.3.7, 5.3.3
- **Browser & Version**: Chrome 88, Firefox 86 and many others
- **OS & Version**: Win10, Linux, Mac OS Big Sur 11.1 
- **Running Example**: https://www.pixiplayground.com/#/edit/egnGEqCLAen4GkMZcYrE2  
- **Example Code**:https://pastebin.com/kMC90VyU

**Comment by bigtimebuddy** [comment]
Thanks for reporting @ps-lvuksan this indeed looks like a regression in behavior.

I tried changing the `cap` and `join` and `miterLimit` properties of the `lineStyle` with not effect. 

I'm not seeing this in 5.2.5, so I think this was introduced in 5.3.0 when we added those new lineStyle Graphics features. @SukantPal this one is probably for you.

**Comment by SukantPal** [comment]
@ps-lvuksan This issue seems to be fixed on PixiJS 6.0.1; I've updated that playground you've posted in this issue and now the bug isn't there.

**Comment by SukantPal** [comment]
This might've been a duplicate of #6722.
